src/sagas/projects.js

```diff
@@ -47,6 +50,17 @@ export function* changeCurrentProject() {
   yield call(saveCurrentProject, state);
 }
 
+export function* rehydrateProject({payload: {rehydratedProject}}) {
+  const state = yield select();
+  const allProjectKeys = getProjectKeys(state);
+  if (allProjectKeys.includes(rehydratedProject.projectKey)) {
+    yield put(__changeCurrentProject(rehydratedProject.projectKey));
+  } else {
+    yield put(projectLoaded(rehydratedProject));
+    yield put(__changeCurrentProject(rehydratedProject.projectKey));
```

Comment:
Generally I consider it a red flag any time more than one action is called from the same place in the code. I try to follow a pattern where actions simply encode _what happened_ (typically either the user took some action, or an asynchronous operation has completed); then reducers and sagas decide what to do about it. In this case I would suggest an e.g. `PROJECT_RESTORED_FROM_LAST_SESSION` action, equivalent to `SNAPSHOT_IMPORTED` and `GIST_IMPORTED`, and then update the appropriate reducers (and their tests!)